src/io/inst_vector.h

```diff
@@ -152,6 +153,58 @@ struct TBlobBatch {
     delete inst_index;
   }
 };  // struct TBlobBatch
+
+class TBlobContainer : public mshadow::TBlob {
```

Comment:
It is used in iter_batchloader.h to replace TensorContainer with DType. With it there is no need to add template DType to the decorative class. You only need to place DType at the inner-most class(ImageRecordIter). Of course there are other ways to achieve this with passing DType all along, which are a little cubersome.